core/json/src/test/scala/net/liftweb/json/ExtractionBugs.scala

```diff
@@ -122,4 +122,50 @@ object ExtractionBugs extends Specification {
     val deserialized = parse(serialized).extract[Holder2]
     deserialized must_== holder
   }
+
+  "deserialize list of homogonous tuples w/ experimental tuples enabled" in {
+    implicit val formats = new DefaultFormats {
+      override val experimentalTupleSupport = true
+    }
+
+    case class Holder(items: List[(String, String)])
+
+    val holder = Holder(List(("string", "string")))
+    val serialized = compactRender(Extraction.decompose(holder))
+
+    val deserialized = parse(serialized).extract[Holder]
+    deserialized must_== holder
+  }
+
+  "deserialize a list of heterogenous tuples w/ experimental tuples enabled" in {
+    implicit val formats = new DefaultFormats {
+      override val experimentalTupleSupport = true
+    }
+
+    // MSF: This currently doesn't work with scala primitives?! The type arguments appear as
+    // java.lang.Object instead of scala.Int. :/
+    case class Holder2(items: List[(String, Integer)])
+
+    val holder = Holder2(List(("string", 10)))
+    val serialized = compactRender(Extraction.decompose(holder))
+
+    val deserialized = parse(serialized).extract[Holder2]
+    deserialized must_== holder
+  }
+
+  "deserialize an out of order old-style tuple w/ experimental tuples enabled" in {
```

Comment:
The old behavior probably isn't covered explicitly, but would be covered in Lift-Json's handling of constructor arguments in ExtractionExamplesSpec.